e2e/integration_tests/data_flow_test.go

```diff
@@ -386,12 +387,43 @@ func Test_VendorList(t *testing.T) {
 		failOnError(err)
 		test_vendor_list = append(test_vendor_list, vendor)
 	}
-	th.Assert(t, len(test_vendor_list)>= len(common_vendor_list), "List of distinct vendors should at least include most common vendors")
+	th.Assert(t, len(test_vendor_list) >= len(common_vendor_list), "List of distinct vendors should at least include most common vendors")
 	Assert.Contains(t, test_vendor_list, common_vendor_list[0], "List of distinct vendors should include Epic")
 	Assert.Contains(t, test_vendor_list, common_vendor_list[1], "List of distinct vendors should include Cerner")
 }
 
 func Test_MetricsAvailableInQuerier(t *testing.T) {
+	var err error
+	qUser := viper.GetString("quser")
+	qPassword := viper.GetString("qpassword")
+	qHost := viper.GetString("qhost")
+	qPort := viper.GetString("qport")
+	qName := viper.GetString("qname")
+
+	hap := th.HostAndPort{Host: qHost, Port: qPort}
+	err = th.CheckResources(hap)
+	if err != nil {
+		panic(err)
+	}
+
+	// Set-up the test queue
+	var mq lanternmq.MessageQueue
+	var chID lanternmq.ChannelID
+	mq, chID, err = aq.ConnectToServerAndQueue(qUser, qPassword, qHost, qPort, qName)
+	defer mq.Close()
+	th.Assert(t, err == nil, err)
+	th.Assert(t, mq != nil, "expected message queue to be created")
+	th.Assert(t, chID != nil, "expected channel ID to be created")
+
+	var wg sync.WaitGroup
+	ctx := context.Background()
+	wg.Add(1)
+	errs := make(chan error)
+
+	// send endpoints so the networkstats querier can read them
+	go se.GetEnptsAndSend(ctx, &wg, qName, 10, store, &mq, &chID, errs)
+	time.Sleep(30 * time.Second)
```

Comment:
thoughts on breaking this out into a helper function?

Reply:
Good idea, I've refactored most of it into helper functions